### PR TITLE
sprinting into railings makes you start vaulting over them

### PR DESCRIPTION
**Pull request by zjdtmkhzt**
## About the PR 
Pretty much what it says in the title.
## Why's this needed? 
Not sure if there is any quicker way already in the game that I am not aware of, but having to click the tiny sprites all the time as rancher is pretty annoying.
## Changelog

```changelog
(u)zjdtmkhzt
(+)Sprinting into railings will make you start the process of vaulting over them.
```

### CODE REVIEW

File: code/obj/railing.dm [hunk 1]
```diff
@@ -144,6 +144,14 @@
...
+		src.try_vault(user)
+
+	Bumped(var/mob/AM as mob)
+		. = ..()
+		if(AM.client?.check_key(KEY_RUN))
+			src.try_vault(AM)
+
+	proc/try_vault(mob/user)
 		if (railing_is_broken(src))
 			user.show_text("[src] is broken! All you can really do is break it down...", "red")
```
Comment (ZeWaka) [inline code/obj/railing.dm:156]: this could get really spammy quick i think if it's broken, test to be sure it's not

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/obj/railing.dm (hunk 1 above)
Comment (zjdtmkhzt) [reply code/obj/railing.dm:156]: Right, good idea! It probably will be.

Comment (zjdtmkhzt) [reply code/obj/railing.dm:156]: Apparently broken railings get density 0, so you can't bump into them. (:

**Review by ZeWaka — APPROVED** [review]
sure

**Comment by Adharainspace** [comment]
how many times is bumped called per sprint into? will it continually try to vault over? im sure you tested this but id like some peace of mind.

**Comment by zjdtmkhzt** [comment]
Continuously running into railings seemed fine to me, it didn't keep restarting the timer or anything.

**Pull request merged**